x/profiles/keeper/msg_server_chain_link.go

```diff
@@ -56,6 +61,10 @@ func (k msgServer) UnlinkChainAccount(goCtx context.Context, msg *types.MsgUnlin
 	// Delete the link
 	k.DeleteChainLink(ctx, link)
 
+	if k.isDefaultExternalAddress(ctx, link) {
+		k.updateOwnerDefaultExternalAddress(ctx, link.User, link.ChainConfig.Name)
+	}
```

Comment:
I think it's better to move this inside the keeper method `DeleteChainLink`

Reply:
The same reason as moving `SaveDefaultExternalAddress` into `SaveChainLink`.

Reply:
I think we should move these lines inside the `DeleteChainLink` anyway. They are private functions that must be called each time a chain link is deleted. Leaving them here would make it hard to deal with any bug that might arise from calling the `DeleteChainLink` method from elsewhere (maybe in an IBC handler or something else). Also, with this feature we are not changing how messages are handled, so we shouldn't change the code here. We should change what is done under the hoods when a chain link is saved/deleted instead